internal/api/core/middleware/proxy.go

```diff
@@ -145,12 +143,23 @@ func (h *httpProxy) serve(c echo.Context) error {
 	req := c.Request()
 	res := c.Response()
 
+	isAllowed := false
+	for i := 0; i < len(h.config.AllowedEndpoints) && !isAllowed; i++ {
+		allowedEndpoint := h.config.AllowedEndpoints[i]
+		isAllowed = isAllowed || (allowedEndpoint.Method == req.Method && len(allowedEndpoint.Endpoint.FindAllString(h.path, -1)) > 0)
```

Comment:
I think you can remove the `isAllowed ||` here because it will always be false anyway (otherwise we don't enter the loop).

Overall, would a normal `range` loop be simpler and clearer?

```go
isAllowed := false
for _, allowedEndpoint := range h.config.AllowedEndpoints {
    if allowedEndpoint.Method == req.Method && len(allowedEndpoint.Endpoint.FindAllString(h.path, -1)) > 0 {
        isAllowed = true
        break;
    }
}
```

Reply:
Hm, `range` loops are way more Go-y and simpler to read to me, even with breaks... but don't feel too strongly either.